server/backend/database/memory/database.go

```diff
@@ -242,12 +242,35 @@ func (d *DB) listAllProjectInfos(
 		return nil, fmt.Errorf("fetch all projects: %w", err)
 	}
 
+	lastIDBytes, err := housekeepingLastProjectID.Bytes()
+	if err != nil {
+		return nil, fmt.Errorf("decode last project id: %w", err)
+	}
+
 	var infos []*database.ProjectInfo
-	for raw := iter.Next(); raw != nil; raw = iter.Next() {
+	for i := 0; i < pageSize; {
+		raw := iter.Next()
+		if raw == nil {
+			*housekeepingLastProjectID = database.DefaultProjectID
+			break
+		}
+
 		info := raw.(*database.ProjectInfo).DeepCopy()
-		infos = append(infos, info)
-	}
 
+		idBytes, err := info.ID.Bytes()
+		if err != nil {
+			return nil, fmt.Errorf("decode project id: %w", err)
+		}
+
+		if bytes.Compare(idBytes, lastIDBytes) > 0 {
```

Comment:
It seems like go-memdb does not provide an ordered result (iterator) for `txn.Get()`... right?
I could not find any reference to the fact that go-memdb provides ordered results.

Reply:
I think you are right. I will add iterator options to sort by id before iterating.